### DISCUSSION TITLE
What's the best audio class for playing .mp3 files in voice channels?

### DISCUSSION THREAD

**Discussion by f1refa11** [Rapptz/discord.py#9256] [Q&A] [answered]
Many tutorials on how to make a music bot use FFmpegPCMAudio class, but in documentation there are 4 audio classes: PCMAudio, FFmpegAudio, FFmpegPCMAudio, FFmpegOpusAudio.
I am going to play .mp3 files in voice channels and would like to know what's the fastest and least CPU consuming class for this.

**Comment by 7468656F** [comment, marked as answer]
Probably FFmpegAudio, since I‘m pretty sure PCM and Opus are lossless audio formats.
